code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -105,68 +109,61 @@ export function composeStory<TRenderer extends Renderer = Renderer, TArgs extend
     abortSignal: new AbortController().signal,
     step: (label, play) => story.runStep(label, play, context),
     canvasElement: globalThis?.document?.body,
-    context: null!,
     canvas: {},
     ...story,
+    context: null!,
+    mount: null!,
   };
 
   context.context = context;
+  context.mount = story.mount(context);
+  context.renderToCanvas = async () => {
+    // Consolidate this renderContext with Context in SB 9.0
+    const unmount = await story.renderToCanvas?.(
+      {
+        componentId: story.componentId,
+        title: story.title,
+        id: story.id,
+        name: story.name,
+        tags: story.tags,
+        showError: (error) => {},
+        showException: (error) => {},
+        forceRemount: true,
+        storyContext: context,
+        storyFn: () => story.unboundStoryFn(context),
+        unboundStoryFn: story.unboundStoryFn,
+      } as RenderContext<TRenderer>,
+      context.canvasElement
+    );
+    if (unmount) {
+      cleanups.push(unmount);
+    }
+  };
 
-  const playFunction = story.playFunction
-    ? async (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
-        Object.assign(context, extraContext);
-        return story.playFunction!(context);
-      }
-    : undefined;
-
-  let previousCleanupsDone = false;
+  const playFunction = (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
+    Object.assign(context, extraContext);
```

Comment:
If the context is mutated in a test, will it leak to another? I remember doing exactly this when working on PS, but then it caused leaking issues between tests

Reply:
fixed